src/workerd/api/sockets.c++

```diff
@@ -179,8 +181,11 @@ jsg::Ref<Socket> Socket::startTls(jsg::Lock& js, jsg::Optional<TlsOptions> tlsOp
     }
   }
   // All non-secure sockets should have a tlsStarter.
-  kj::Own<kj::AsyncIoStream> secure = KJ_ASSERT_NONNULL(*tlsStarter)(acceptedHostname);
-  return setupSocket(js, kj::mv(secure), kj::mv(options), kj::mv(tlsStarter), true, kj::mv(domain));
+  KJ_ASSERT_NONNULL(*tlsStarter)(acceptedHostname);
+
+  return setupSocket(
```

Comment:
Do we maybe want to change the `startTls()` API to modify the existing socket in-place instead of returning a new one, since that's what we've done with the underlying KJ APIs?

Reply:
No, I don't think so. I much prefer having a clean separation between the plaintext socket and the secured socket.